src/sent.rs

```diff
@@ -153,10 +169,49 @@ impl SentPackets {
         self.congestion_ctrl.on_transmit(seq_num, transmit).unwrap();
     }
 
+    /// Handle an ACK for a packet with sequence number `ack_num`, and an optional selective_ack.
+    ///
+    /// Returns Error, if `ack_num` is not within the sequence number range of the sent packets.
+    pub fn on_ack(
+        &mut self,
+        ack_num: u16,
+        selective_ack: Option<&SelectiveAck>,
+        delay: Duration,
+        now: Instant,
+    ) -> Result<(CircularRangeInclusive, Vec<u16>), SentPacketsError> {
+        let range = self.seq_num_range();
+        if range.contains(ack_num) {
+            // Do not ACK if ACK num corresponds to initial packet.
+            if ack_num != range.start() {
+                self.on_full_ack(ack_num, selective_ack, delay, now);
+            }
+
+            // Mark all packets up to `ack_num` acknowledged by retaining all packets in
+            // the range beyond `ack_num`.
+            let full_acked = CircularRangeInclusive::new(range.start(), ack_num);
+
+            let selected_acks = if let Some(selective_ack) = selective_ack {
+                selective_ack
+                    .acked()
+                    .iter()
+                    .enumerate()
+                    .filter(|(_, &acked)| acked)
+                    .map(|(i, _)| ack_num.wrapping_add(i as u16).wrapping_add(2))
```

Comment:
```suggestion
                    .map(|(i, _)| ack_num.wrapping_add(2).wrapping_add(i as u16))
```

Reply:
What is helpful about reordering it this way?